src/app.service.ts

```diff
@@ -46,105 +68,120 @@ export class AppService {
       .post(
         `${this.baseUrl}/gateway/initialize`,
         {
-          total_amount:newTransaction.total_amount,
-          transaction_id:newTransaction.transaction_id,
-          currency:newTransaction.currency,
-          return_url:newTransaction.return_url
+          total_amount: newTransaction.total_amount,
+          transaction_id: transaction_id,
+          currency: currency,
+          return_url: return_url,
         },
         {
           headers: {
             'Content-Type': 'application/json',
-            'Authorization': `Basic ${base64data}`,
+            Authorization: `Basic ${base64data}`,
             'x-api-Key': process.env.apiKey,
-            'mode': process.env.mode,
+            mode: process.env.mode,
           },
         },
       )
       .then((request) => {
-        let storeProduct = request.data.data
-      storeProduct = new this.getTransactionModel({
-        t_id:request.data.data.t_id,
-        t_sum:request.data.data.t_sum,
-        t_url:request.data.data.t_url,
-          transaction_url:request.data.data.transaction_url,
-          transaction_id:request.data.data.transaction_id,
-        },)
-        console.log(request.data)
-        return   storeProduct.save()
+        let storeProduct = request.data.data;
+        storeProduct = new this.getTransactionModel({
+          t_id: request.data.data.t_id,
+          t_sum: request.data.data.t_sum,
+          t_url: request.data.data.t_url,
+          transaction_url: request.data.data.transaction_url,
+          transaction_id: request.data.data.transaction_id,
+        });
+        console.log(request.data);
+        storeProduct.save();
       })
       .catch((error) => {
         console.log(error);
       });
-
   }
- public async getTransact(_id){
-   let transaction:any;
-   transaction = await this.getTransactionModel.findById(_id);
-   console.log('transaction here', transaction);
+  public async getTransact(_id) {
+    let transaction: any;
+    // eslint-disable-next-line prefer-const
+    transaction = await this.getTransactionModel.findById(_id);
+    console.log('transaction here', transaction);
     const appBasic = `${process.env.apiSecret}`;
     const buff = Buffer.from(appBasic);
     const base64data = buff.toString('base64');
     console.log(base64data);
-     await axios
-      .get(`${this.baseUrl}/gateway/gateways?t_url="${transaction.t_url}"&t_id="${transaction.t_id}"&t_sum="${transaction.t_sum}"`, {
-        headers: {
-          'Content-Type': 'application/json',
-          Authorization: `Basic ${base64data}`,
-          'x-api-Key': process.env.apiKey,
-          mode: process.env.mode,
-        }
```

Comment:
This is redundancy and it eats memory space. Please resolve this as I told you earlier.